pkg/objstore/swift/swift.go

```diff
@@ -0,0 +1,314 @@
+	c.opsTotal.WithLabelValues(opObjectsList).Inc()
+	// Ensure the object name actually ends with a dir suffix. Otherwise we'll just iterate the
+	// object itself as one prefix item.
+	if dir != "" {
+		dir = strings.TrimSuffix(dir, DirDelim) + DirDelim
+	}
+
+	options := &objects.ListOpts{Full: false, Prefix: dir, Delimiter: DirDelim}
+	err := objects.List(c.client, c.name, options).EachPage(func(page pagination.Page) (bool, error) {
+		objectNames, err := objects.ExtractNames(page)
+		if err != nil {
+			return false, err
+		}
+		for _, objectName := range objectNames {
+			if err := f(objectName); err != nil {
+				return false, err
+			}
+		}
+
+		return true, nil
+	})
+
+	return err
+}
+
+// Get returns a reader for the given object name.
+func (c *Container) Get(ctx context.Context, name string) (io.ReadCloser, error) {
+	if name == "" {
+		return nil, fmt.Errorf("error, empty container name passed")
+	}
+	c.opsTotal.WithLabelValues(opObjectGet).Inc()
+	response := objects.Download(c.client, c.name, name, nil)
+	return response.Body, response.Err
+}
+
+// GetRange returns a new range reader for the given object name and range.
+func (c *Container) GetRange(ctx context.Context, name string, off, length int64) (io.ReadCloser, error) {
+	c.opsTotal.WithLabelValues(opObjectGet).Inc()
+	options := objects.DownloadOpts{
+		Newest: true,
+		Range:  fmt.Sprintf("bytes=%d-%d", off, off+length-1),
+	}
+	response := objects.Download(c.client, c.name, name, options)
+	return response.Body, response.Err
+}
+
+// Exists checks if the given object exists.
+func (c *Container) Exists(ctx context.Context, name string) (bool, error) {
+	c.opsTotal.WithLabelValues(opObjectGet).Inc()
+	err := objects.Get(c.client, c.name, name, nil).Err
+	if err == nil {
+		return true, nil
+	}
+
+	if _, ok := err.(gophercloud.ErrDefault404); ok {
+		return false, nil
+	}
+
+	return false, err
+}
+
+// IsObjNotFoundErr returns true if error means that object is not found. Relevant to Get operations.
+func (c *Container) IsObjNotFoundErr(err error) bool {
+	_, ok := err.(gophercloud.ErrDefault404)
+	return ok
+}
+
+// Upload writes the contents of the reader as an object into the container.
+func (c *Container) Upload(ctx context.Context, name string, r io.Reader) error {
+	c.opsTotal.WithLabelValues(opObjectInsert).Inc()
+
+	options := &objects.CreateOpts{Content: r}
+	res := objects.Create(c.client, c.name, name, options)
+	return res.Err
+}
+
+// Delete removes the object with the given name.
+func (c *Container) Delete(ctx context.Context, name string) error {
+	c.opsTotal.WithLabelValues(opObjectDelete).Inc()
+	return objects.Delete(c.client, c.name, name, nil).Err
+
+}
+
+func (*Container) Close() error {
+	// nothing to close
+	return nil
+}
+
+func (c *Container) CreateContainer(name string) error {
+	return containers.Create(c.client, name, nil).Err
+}
+
+func (c *Container) DeleteContainer(name string) error {
```

Comment:
Does `Delete|Create` container need to be exported?

Reply:
updated.